pkg/actor/actor_test.go

```diff
@@ -14,46 +14,168 @@ See the License for the specific language governing permissions and
+	"github.com/cockroachdb/cockroach-operator/pkg/utilfeature"
+	"github.com/google/go-cmp/cmp"
 	"github.com/stretchr/testify/require"
-	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+	"testing"
 )
 
-func TestActorOrder(t *testing.T) {
+func containsAction(actions []api.ActionType, action api.ActionType) bool {
+	for _, a := range actions {
+		if a == action {
+			return true
+		}
+	}
+	return false
+}
 
-	// So the actor order is so important that we need to ensure that the slice
-	// does not get touched.
-	// This test ensures that the name of the actor is in the correct order.
+func TestDecommissionFeatureGate(t *testing.T) {
+	// Setup fake client
+	cluster := testutil.NewBuilder("cockroachdb").
+		Namespaced("default").
+		WithUID("cockroachdb-uid").
+		WithPVDataStore("1Gi", "standard" /* default storage class in KIND */).
+		WithNodeCount(1).Cluster()
+	director := actor.ClusterDirector{}
 
+	cluster.SetTrue(api.InitializedCondition)
+
+	utilfeature.DefaultMutableFeatureGate.Set("UseDecommission=true")
+	actions := director.GetActionsToExecute(cluster)
+	require.True(t, containsAction(actions, api.DecommissionAction))
+
+	utilfeature.DefaultMutableFeatureGate.Set("UseDecommission=false")
+	actions = director.GetActionsToExecute(cluster)
+	require.False(t, containsAction(actions, api.DecommissionAction))
+}
+
+func TestVersionValidatorFeatureGate(t *testing.T) {
 	// Setup fake client
-	builder := fake.NewClientBuilder()
+	cluster := testutil.NewBuilder("cockroachdb").
+		Namespaced("default").
+		WithUID("cockroachdb-uid").
+		WithPVDataStore("1Gi", "standard" /* default storage class in KIND */).
+		WithNodeCount(1).Cluster()
+	director := actor.ClusterDirector{}
 
-	client := builder.Build()
-	actors := NewOperatorActions(nil, client, nil)
+	cluster.SetTrue(api.InitializedCondition)
 
-	actorNamesInOrder := []string{
-		"*actor.decommission",
-		"*actor.versionChecker",
-		"*actor.generateCert",
-		"*actor.partitionedUpdate",
-		"*actor.resizePVC",
-		"*actor.deploy",
-		"*actor.initialize",
-		"*actor.clusterRestart",
-	}
+	utilfeature.DefaultMutableFeatureGate.Set("CrdbVersionValidator=true")
+	actions := director.GetActionsToExecute(cluster)
+	require.True(t, containsAction(actions, api.VersionCheckerAction))
 
-	// TODO this seems kinda hacky, but I could not get a.(type) working
-	// so I am just testing the name of the string. Maybe I could test the
-	// struct type, but hey this works for now.
-	for i, a := range actors {
-		xType := reflect.TypeOf(a)
-		expected := actorNamesInOrder[i]
-		require.True(t, xType.String() == expected, "expected:", expected, "got", xType.String())
-	}
+	utilfeature.DefaultMutableFeatureGate.Set("CrdbVersionValidator=false")
+	actions = director.GetActionsToExecute(cluster)
+	require.False(t, containsAction(actions, api.VersionCheckerAction))
+}
+
+func TestResizePVCFeatureGate(t *testing.T) {
+	// Setup fake client
+	cluster := testutil.NewBuilder("cockroachdb").
```

Comment:
Worth extracting a `createTestCluster` func?

Reply:
Another option might be to combine these tests in a table driven test like `TestDirectorGetActionsToExecute`. Then the cluster and director could be shared for all of them.